src/server/scripts/Pandaria/zone_the_wandering_isle.cpp

```diff
@@ -0,0 +1,2372 @@
+    {
+        if (player->IsAlive() && player->IsVehicle() && player->GetQuestStatus(QUEST_NONE_LEFT_BEHIND) == QUEST_STATUS_INCOMPLETE)
+        {
+            if (entered)
+            {
+                if (Creature* creature = player->FindNearestCreature(NPC_INJURED_SAILOR, 1.0f, true))
+                    creature->CastSpell(player, SPELL_CANCEL_RESCUE_AURA, true);
+
+                return true;
+            }
+        }
+        return false;
+    }
+};
+
+enum AysaVordrakaFightEvents
+{
+    EVENT_TEMPERED_FURY         = 1,
+    EVENT_COMBAT_ROLL           = 2,
+    EVENT_UPDATE_PHASES         = 3
+};
+
+enum AysaVordrakaFightSpells
+{
+    SPELL_TEMPERED_FURY         = 117275,
+    SPELL_COMBAT_ROLL           = 117312,
+    SPELL_FORCECAST_SUMMON_AYSA = 117499,
+    SPELL_DEEP_SEA_SMASH        = 117287
+};
+
+enum AysaVordrakaFightData
+{
+    DATA_AYSA_TALK_INTRO        = 1,
+    DATA_AYSA_TALK_SMASH        = 2,
+    DATA_SUMMON_AGGRESSORS      = 3,
+    DATA_VORDRAKA_DEATH         = 4,
+};
+
+enum AysaVordrakaFightTexts
+{
+    TEXT_INTRO                  = 0,
+    TEXT_SMASH                  = 1,
+    TEXT_REINFORCEMENTS         = 2,
+    TEXT_LOW_HP                 = 3,
+    TEXT_DEATH                  = 4
+};
+
+enum AysaVordrakaFightMisc
+{
+    QUEST_AN_ANCIENT_EVIL       = 29798,
+    QUEST_RISKING_IT_ALL        = 30767,
+    NPC_VORDRAKA                = 56009,
+    DB_PHASE_FIGHT              = 543,
+    DB_PHASE_AFTER_FIGHT        = 993
+};
+
+class npc_aysa_vordraka_fight : public CreatureScript
+{
+public:
+    npc_aysa_vordraka_fight() : CreatureScript("npc_aysa_vordraka_fight") { }
+
+    struct npc_aysa_vordraka_fightAI : public ScriptedAI
+    {
+        npc_aysa_vordraka_fightAI(Creature* creature) : ScriptedAI(creature) {
+            me->SetReactState(REACT_DEFENSIVE);
+        }
+
+        void MoveInLineOfSight(Unit* who) override
+        {
+            if (who->IsPlayer())
+                if (Creature* creature = me->FindNearestCreature(NPC_VORDRAKA, me->GetVisibilityRange(), true))
+                    AttackStart(creature);
+        }
+
+        void EnterCombat(Unit* who) override
+        {
+            who->GetAI()->AttackStart(me);
+            _events.ScheduleEvent(EVENT_TEMPERED_FURY, urand(2000, 4000));
+        }
+
+        void DamageTaken(Unit* /*attacker*/, uint32& damage) override
+        {
+            if (HealthAbovePct(85))
+                damage = urand(1, 2);
+            else
+                me->SetHealth(me->GetMaxHealth() * 0.85f);
+        }
+
+        void SpellHit(Unit* /*caster*/, SpellInfo const* spell) override
+        {
+            if (spell->Id == SPELL_DEEP_SEA_SMASH)
+            {
+                _events.CancelEvent(EVENT_TEMPERED_FURY);
+                _events.ScheduleEvent(EVENT_COMBAT_ROLL, 1000);
+            }
+        }
+
+        void SetData(uint32 id, uint32 /*value*/) override
+        {
+            switch (id)
```

Comment:
Can't you just call `Talk` from `creature->AI()` instead of resorting to `SetData` ? If not, then `DoAction` is more appropriate for this